File: docs/guides/java/src/test/java/com/algolia/setHeaderUserIDThenSaveObjects.java
package com.algolia;

import com.algolia.api.SearchClient;
import com.algolia.config.*;
import com.algolia.model.search.*;
import java.util.List;
import java.util.Map;

public class setHeaderUserIDThenSaveObjects {

  private static final List<Map<String, Object>> playlists = List.of(/* Your records */);

  public static void main(String[] args) throws Exception {
    try (SearchClient client = new SearchClient("ALGOLIA_APPLICATION_ID", "ALGOLIA_API_KEY");) {
      playlists.forEach(playlist -> {
        String playlistUserID = (String) playlist.get("userID");
        client.saveObjects(
          "<YOUR_INDEX_NAME>",
          playlists,
          false,
          1000,
          new RequestOptions().addExtraHeader("X-Algolia-User-ID", playlistUserID)
        );
      });
    } catch (Exception e) {
      System.out.println("An error occurred: " + e.getMessage());
    }
  }
}

File: docs/guides/kotlin/src/main/kotlin/com/algolia/snippets/setHeaderUserIDThenSaveObjects.kt
import kotlinx.serialization.json.JsonObject

import com.algolia.client.api.SearchClient
import com.algolia.client.configuration.*
import com.algolia.client.transport.*
import com.algolia.client.extensions.*
import com.algolia.client.model.search.*

suspend fun setHeaderUserIDThenSaveObjects() {
  val playlists: List<JsonObject> = listOf() // Your records

  val client = SearchClient(appId = "ALGOLIA_APPLICATION_ID", apiKey = "ALGOLIA_API_KEY")

  playlists.forEach { playlist ->
    val playlistUserID = playlist["userID"].toString()
    try {
      client.saveObjects(
        indexName = "<YOUR_INDEX_NAME>",
        objects = playlists,
        waitForTasks = false,
        batchSize = 1000,
        requestOptions = RequestOptions(
          headers = buildMap {
            put("X-Algolia-User-ID", playlistUserID)
          },
        ),
      )
    } catch (exception: Exception) {
      println(exception.message)
    }
  }
}
